### PR TITLE
Improve hybridblock doc

### PR DESCRIPTION
**Pull request by piiswrong**
## Description ##
(Brief description on what this PR is about)

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: docs/tutorials/gluon/hybrid.md [hunk 1]
```diff
@@ -105,23 +105,35 @@ Hybridize will speed up execution and save memory. If the top level layer is
 not a `HybridBlock`, you can still call `.hybridize()` on it and Gluon will try
 to hybridize its children layers instead.
 
+`hybridize` also accepts many options for performance tuning. For example, you
```
Comment (szha) [inline docs/tutorials/gluon/hybrid.md]: many -> several?

File: docs/tutorials/gluon/hybrid.md [hunk 2]
```diff
@@ -105,23 +105,35 @@ Hybridize will speed up execution and save memory. If the top level layer is
...
+can do
+
+```python
+net.hybridize(static_alloc=True)
+# or
+net.hybridize(static_alloc=True, static_shape=True)
+```
+
+Please refer to [API manual](https://mxnet.incubator.apache.org/api/python/gluon/gluon.html?highlight=hybridize#mxnet.gluon.Block.hybridize)
```
Comment (ThomasDelteil) [inline docs/tutorials/gluon/hybrid.md]: to the* API manual

File: python/mxnet/gluon/block.py [hunk 3]
```diff
@@ -600,6 +605,28 @@ def _summary_hook(block, _, outputs):
 class HybridBlock(Block):
     """`HybridBlock` supports forwarding with both Symbol and NDArray.
 
+    `HybridBlock` is similar to `Block`, with a few differencies::
```
Comment (ThomasDelteil) [inline python/mxnet/gluon/block.py]: differences*

File: python/mxnet/gluon/block.py [hunk 4]
```diff
@@ -600,6 +605,28 @@ def _summary_hook(block, _, outputs):
...
+                return F.relu(self.dense1(x))
+
+        model = Model()
+        model.initialize(ctx=mx.cpu(0))
+        model.hybridize()
+        model(mx.nd.zeros((10, 10), ctx=mx.cpu(0)))
+
     Forward computation in :py:class:`HybridBlock` must be static to work with :py:class:`Symbol` s,
     i.e. you cannot call :py:meth:`NDArray.asnumpy`, :py:attr:`NDArray.shape`,
     :py:attr:`NDArray.dtype`, etc on tensors.
```
Comment (ThomasDelteil) [inline python/mxnet/gluon/block.py]: NDArray-style indexing wouldn't work either in an HybridBlock. I think it would be worth mentioning it.
